open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/test/java/com/wso2/openbanking/accelerator/consent/extensions/manage/vrp/VRPConsentRequestValidatorTest.java

```diff
@@ -0,0 +1,2630 @@
+
+        JSONObject isValid = VRPConsentRequestValidator.validatePeriodicLimits(invalidLimit);
+
+        Assert.assertFalse((Boolean) isValid.get(ConsentExtensionConstants.IS_VALID));
+        Assert.assertEquals(ErrorConstants.MISSING_PERIOD_LIMITS,
+                isValid.get(ConsentExtensionConstants.ERRORS));
+    }
+
+
+    @Test
+    public void testValidateParameterDateTime_InvalidDateTimeRange() {
+        // Arrange
+        JSONObject controlParameters = new JSONObject();
+        controlParameters.put(ConsentExtensionConstants.VALID_FROM_DATE_TIME, "2022-01-01T00:00:00Z");
+        controlParameters.put(ConsentExtensionConstants.VALID_TO_DATE_TIME, "2022-01-02T00:00:00Z");
+        JSONObject result1 = VRPConsentRequestValidator.validateParameterDateTime(controlParameters);
+        Assert.assertNotNull(result1);
+        Assert.assertFalse((Boolean) result1.get(ConsentExtensionConstants.IS_VALID));
+    }
+
+
+    @Test
+    public void testDateValidation() {
+        // Arrange
+        JSONObject controlParameters = new JSONObject();
+        controlParameters.put(ConsentExtensionConstants.VALID_FROM_DATE_TIME, "2022-01-01T00:00:00Z");
+        controlParameters.put(ConsentExtensionConstants.VALID_TO_DATE_TIME, "2023-01-01T00:00:00Z");
+
+        JSONObject validationResult = VRPConsentRequestValidator.validateParameterDateTime(controlParameters);
+
+        // Assert
+        Assert.assertFalse((Boolean) validationResult.get(ConsentExtensionConstants.IS_VALID));
+
+        // Negative scenario: ValidToDateTime is older than the current date
+        controlParameters.put(ConsentExtensionConstants.VALID_TO_DATE_TIME, "2021-01-01T00:00:00Z");
+        validationResult = VRPConsentRequestValidator.validatePeriodicLimits(controlParameters);
+        Assert.assertFalse((Boolean) validationResult.get(ConsentExtensionConstants.IS_VALID));
+
+        // Negative scenario: CurrentDate is older than ValidFromDateTime
+        controlParameters.put(ConsentExtensionConstants.VALID_TO_DATE_TIME, "2023-01-01T00:00:00Z");
+        controlParameters.put(ConsentExtensionConstants.VALID_FROM_DATE_TIME, "2024-01-01T00:00:00Z");
+        validationResult = VRPConsentRequestValidator.validateParameterDateTime(controlParameters);
+        Assert.assertFalse((Boolean) validationResult.get(ConsentExtensionConstants.IS_VALID));
+    }
+
+
+    @Test
+    public void testInvalidDateTimeRange() {
+
+        JSONObject controlParameters = new JSONObject();
+        controlParameters.put(ConsentExtensionConstants.VALID_FROM_DATE_TIME, "2023-01-01T00:00:00Z");
+        controlParameters.put(ConsentExtensionConstants.VALID_TO_DATE_TIME, "2022-01-01T00:00:00Z");
+
+        boolean hasValidFromDate = controlParameters.containsKey(ConsentExtensionConstants.VALID_FROM_DATE_TIME);
+        boolean hasValidToDate = controlParameters.containsKey(ConsentExtensionConstants.VALID_TO_DATE_TIME);
+
```

Comment:
Check for additional empty line here as well

Reply:
Fixed in [876c68f]